package/gluon-core/files/usr/lib/lua/gluon/util.lua

```diff
@@ -81,12 +82,25 @@ end
 -- (4, 0): mesh VPN
 -- (5, n): mesh interface for n'th radio (802.11s)
 function generate_mac(f, i)
-  local m1, m2, m3, m4, m5, m6 = string.match(sysconfig.primary_mac, '(%x%x):(%x%x):(%x%x):(%x%x):(%x%x):(%x%x)')
-  m1 = nixio.bit.bor(tonumber(m1, 16), 0x02)
-  m2 = (tonumber(m2, 16)+f) % 0x100
-  m3 = (tonumber(m3, 16)+i) % 0x100
+  local hashed = string.sub(hash.md5(sysconfig.primary_mac), 0, 12)
+  local m1, m2, m3, m4, m5, m6 = string.match(hashed, '(%x%x)(%x%x)(%x%x)(%x%x)(%x%x)(%x%x)')
 
-  return string.format('%02x:%02x:%02x:%s:%s:%s', m1, m2, m3, m4, m5, m6)
+  m1 = tonumber(m1, 16)
+  m3 = tonumber(m3, 16)
+  m6 = tonumber(m6, 16)
+
+  m1 = nixio.bit.bor(m1, 0x02)  -- set locally administered bit
+  m1 = nixio.bit.band(m1, 0xFE) -- unset the multicast bit
+  m3 = (m3+i) % 0x100           -- add interface identifier
```

Comment:
Also incrementing the third byte will yield a higher collision change than the scheme suggested in #648, as each node will use two or three instead of one 45bit prefix.

I'd like to see this function to be changed to use only a single parameter *i* which defines the index, and all uses of it to be adjusted according to #648.